### PR TITLE
Make CRDs Helm 3 compatiable

### PR DESCRIPTION
**Pull request by yurinnick**
- Remove `"helm.sh/hook": crd-install` from `crds/*.yaml`
- Copy Helm v2 CRDs to `templates/crds`, gated by `installCRDs` value

Based on [argo-cd](https://github.com/argoproj/argo-helm/tree/master/charts/argo-cd) helm chart.

Closes https://github.com/vmware-tanzu/helm-charts/issues/7.

### CODE REVIEW

**Comment by yurinnick** [comment]
Suppose to fix CRDs creation for Helm 3 #7

**Comment by jenting** [comment]
I was wondering if it possible we could do removing `"helm.sh/hook": crd-install` from `crds/*.yaml` without copy duplicated CRDs to `templates/crds/*.yaml` to make it compatible on both helmv2 and helmv3.

BTW, I think set `installCRDs=false` when helm install would cause velero server install failed because when velero server startup, velero tries to access the velero CRDs resources.

**Comment by yurinnick** [comment]
I don't this there is a way to keep backward compatibility without having a copy of CRDs with a `crd-install` hook. Why Helms team decided to deprecate this hook in such a breaking way is beyond me.

For Helm v3 `installCRDs=false` suppose to keep templated CRDs with `crd-install` hook from installing, so they won't break a deployment.

**Comment by yurinnick** [comment]
@carlisia this seems to be the critical functionality for many people using Helm v3. Is there any chance to get it merge in nearest future?

**Comment by carlisia** [comment]
Hey @yurinnick I missed this PR! I will review it later today, and thank you!

File: charts/velero/Chart.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ apiVersion: v1
 appVersion: 1.3.0
 description: A Helm chart for velero
 name: velero
-version: 2.9.3
+version: 2.9.4
```
Comment (cpanato) [inline charts/velero/Chart.yaml]: please rebase your PR and bump the chart version

**Comment by yurinnick** [comment]
Blocked by #85

**Comment by carlisia** [comment]
I'm still testing this. So far:

- getting the `manifest_sorter.go:192: info: skipping unknown hook: "crd-install"`. Investigating if we could make it so we don't output this, maybe change how the hook is handled.
- the velero deployment is not being created. I'm looking to see if it's something on my end.

**Comment by yurinnick** [comment]
@carlisia by default installation uses Helm v2 approach of imstalling CRDs from templates. To enable Helm v3 "mode", please set `installCRDs: false` in `values.yaml`. Should I enable Helm v3 "mode" by default?

**Comment by carlisia** [comment]
Ah, thank you! Let me review the whole thing and see what changes if any we should make.

**Comment by yurinnick** [comment]
Okay, so with `ct` using Helm v3 by default I disabled templated CRDs installation for CI. Also, I gated `cleanup-cdrs` with all other Helm v2 CRDs templates, so it won't explode CI builds on cleanup.

**Comment by carlisia** [comment]
Should `cleanUpCRDs` exist in the `values.yaml`?

**Comment by yurinnick** [comment]
Well, for Helm v3 deployments including CI it doesn't matter now and it worked for Helm v2 before, so... I guess it's fine 😃 . I will add default value for in into `values.yaml` though.

**Comment by carlisia** [comment]
I'm able to install using Helm 3. But my restore is not restoring as I expected. Investigating. 

Still need to check compatibility with Helm 2.

**Comment by carlisia** [comment]
- [ ] backup w/ Heml 3
- [ ] restore w/ Heml 3
- [ ] backup w/ Heml 2
- [ ] restore w/ Heml 2

**Comment by yurinnick** [comment]
Could you provide steps to reproduce as well as error messages if you got any?

**Comment by carlisia** [comment]
No, I got it to pass (see the checkboxes above). Today I'll test the chart with Helm 2.

**Comment by gadelkareem** [comment]
@carlisia IMHO if someone needs Helm 2 then they can just use an older chart with a diff version in the values

**Comment by carlisia** [comment]
@nrb / @cpanato / @skriss PTAL when you have a chance.

I volunteer to add a new PR with an update to the documentation.

**Comment by yurinnick** [comment]
@nrb @cpanato could you review please?

**Comment by cprivite** [comment]
Does README.md need to be updated to tell people to supply installCRDs: false for helm 3 installs?

**Comment by carlisia** [comment]
@cprivite yes, that's probably a good idea. Would you like to add it?

**Pull request merged**